### DISCUSSION TITLE
json syntax error on first launch of csv-importer: docker-compose

### DISCUSSION THREAD

**Discussion by princinv** [firefly-iii/firefly-iii#4429] [Firefly III]
I am running Firefly III version 5.4.6.

**Description**
Congratulations on my first GitHub support post!  While i can successfully get the firefly program running on its own, i am struggling with the official csv-importer. The importer webUI does come up correctly, but regardless of whether i specify an access token or not, specify my registered domain or container name or not, i recieve the same error message:

Could not decode JSON from URL "https://fireflyiii.[redacted]/api/v1/about": Syntax error

I would include more detailed logs if i were able to find any which relate to this error, alas i cannot. I will however supply my docker-compose.yml file: ($$ sections are purely relics from previous efforts to fix the issue, and remain until i solve the issue because i have flashbacks to TypeToLearn everytime i retype something i've already typed before)

version: '3.3'

services:
  fireflyiii:
    container_name: fireflyiii
    image: jc5x/firefly-iii:latest
    volumes:
      - /home/[redacted]/docker/fireflyiii:/var/www/html/storage/upload
    env_file: .env
$$    ports:
$$      - 8081:8080
    depends_on:
      - fireflyiii_db
  fireflyiii_db:
    container_name: fireflyiii_db
    image: yobasystems/alpine-mariadb:latest
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=yes
      - MYSQL_USER=[redacted]
      - MYSQL_PASSWORD=[redacted]
      - MYSQL_DATABASE=fireflyiii_db
    volumes:
      - /home/[redacted]/docker/fireflyiii/mariadb_fireflyiii:/var/lib/mysql
  fireflyiii_importer:
    container_name: fireflyiii_importer
    image: fireflyiii/csv-importer:latest
$$    environment:
$$      - FIREFLY_III_ACCESS_TOKEN=[redacted]
$$      - FIREFLY_III_URL=[redacted]
$$    volumes:
$$      - $PWD:/import
    ports:
      - 8082:8080
volumes:
   firefly_iii_upload:
   firefly_iii_db:

networks:
  default:
    external:
      name: lsio

**Extra info**
Some background on my setup: I have migrated my self-hosted stack to a new machine and dockerized all processes. Created following (mostly) linuxserver.io and perfectmediaserver.com guides, firefly's webUI is routed through the linuxserver SWAG (nginx) reverse proxy. Firefly and the database and importer operate within their own docker-compose file on the same default network as the rest of my containers.

**Edits**
Getting the following error in logs when i attempt to authenticate with Client_ID using the reauthenticate button on the importer webui:

2021-02-25  2:03:22 55 [ERROR] InnoDB: Table `fireflyiii_db`.`oauth_access_tokens` is corrupted. Please drop the table and recreate.

And this error (in browser) when the client ID is entered into the webui:

SQLSTATE[42S02]: Base table or view not found: 1932 Table 'fireflyiii_db.oauth_access_tokens' doesn't exist in engine (SQL: select * from `oauth_access_tokens` where `oauth_access_tokens`.`client_id` = 8 and `oauth_access_tokens`.`client_id` is not null and `user_id` = 1 and `revoked` = 0 and `expires_at` > 2021-02-24 19:14:26 order by `expires_at` desc limit 1) 

I have completely removed the databases and recreated the container several times but these error crop up at many different places...permissions maybe?

**Comment by ColinMaudry** [comment]
I had the same issue, and the reason was the target URL was not accessible. To debug it, I did:

```
curl -LI -H "Accept: application/json" [FIREFLY_URL]/api/v1/about
```

In my case, it redirected to a login page, although the Access Token had been accepted bu the importer.
